services/venice-controller/src/main/java/com/linkedin/venice/controller/stats/AddVersionLatencyStats.java

```diff
@@ -0,0 +1,104 @@
+package com.linkedin.venice.controller.stats;
+
+import com.linkedin.venice.stats.AbstractVeniceStats;
+import io.tehuti.metrics.MetricsRepository;
+import io.tehuti.metrics.Sensor;
+import io.tehuti.metrics.stats.Avg;
+import io.tehuti.metrics.stats.Max;
+
+
+/**
+ * This class is used to track the latency of various operations related to adding a version in Venice.
+ * <p>
+ * The following sensors are tracked:
+ * <ul>
+ *   <li><b>existingSourceVersionHandlingLatencySensor</b> –
+ *       Measures latency when handling add version requests with an existing source version.</li>
+ *   <li><b>batchTopicCreationLatencySensor</b> –
+ *       Tracks latency for creating batch topics (used by both child and parent controllers).</li>
+ *   <li><b>resourceAssignmentWaitLatencySensor</b> –
+ *       Captures time spent waiting for node resource assignments.</li>
+ *   <li><b>helixResourceCreationLatencySensor</b> –
+ *       Monitors the latency of creating Helix storage cluster resources.</li>
+ *   <li><b>retireOldVersionsLatencySensor</b> –
+ *       Records the time taken to retire outdated store versions.</li>
+ *   <li><b>startOfPushLatencySensor</b> –
+ *       Measures latency for sending the start-of-push signal.</li>
+ *   <li><b>versionCreationFailureLatencySensor</b> –
+ *       Tracks latency during version creation failure handling.</li>
+ * </ul>
+ * Each metric uses milliseconds as the unit of measurement.
+ */
+public class AddVersionLatencyStats extends AbstractVeniceStats {
+  /** Measures the time taken to retire outdated store versions. */
+  private final Sensor retireOldVersionsLatencySensor;
+
+  /** Captures the latency while waiting for node resource assignments. */
+  private final Sensor resourceAssignmentWaitLatencySensor;
+
+  /** Tracks latency during the handling of version creation failures. */
+  private final Sensor versionCreationFailureLatencySensor;
+
+  /** Measures latency for add version requests where the source version already exists. */
+  private final Sensor existingSourceVersionHandlingLatencySensor;
+
+  /** Records the time taken to send the start-of-push signal. */
+  private final Sensor startOfPushLatencySensor;
+
+  /** Tracks latency for creating batch topics, applicable to both parent and child controllers. */
+  private final Sensor batchTopicCreationLatencySensor;
+
+  /** Monitors the time required to create Helix storage cluster resources. */
+  private final Sensor helixResourceCreationLatencySensor;
+
+  public AddVersionLatencyStats(MetricsRepository metricsRepository, String name) {
+    super(metricsRepository, name);
+    retireOldVersionsLatencySensor =
+        registerSensorIfAbsent("add_version_retire_old_versions_latency", new Avg(), new Max());
```

Comment:
Should we use percentiles instead?

Reply:
I have added percentile metrics `Generate a histogram stat that emits P50, P95, and P99 values.`. I still keep the avg, and max in case we want to compare with the base metric. Let me know your thoughts!